qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -577,7 +577,12 @@ def _basis_search(equiv_lib, source_basis, target_basis):
 
     # we add a dummy node and connect it with gates in the target basis.
     # we'll start the search from this dummy node.
-    dummy = graph.add_node(NodeData(key="key", equivs=[("dummy starting node", 0)]))
+    dummy = graph.add_node(
+        NodeData(
+            key=Key("key", 0),
```

Comment:
I can see how it can potentially cause an issue. Should I maybe use a really big number close to infinity as the number of qubits in this case?

Reply:
I was thinking of some sort of generated string rather than "key". But changing the integer is probably fine. We probably want to avoid exceeding the size of the appropriate machine integer. $2^{31} - 1$ is the maximum value of `i32`.